src/http/error.rs

```diff
@@ -19,40 +17,10 @@ pub struct DiscordJsonError {
     pub message: String,
     /// The full explained errors with their path in the request
     /// body.
+    #[serde(deserialize_with = "deserialize_errors")]
```

Comment:
This should probably call `Default::default` when the list of errors is absent, as suggested by the `unwrap_or_default` call in the manual deserialisation code.

```suggestion
    #[serde(default, deserialize_with = "deserialize_errors")]
```